modules/custom/activity_basics/activity_basics.module

```diff
@@ -8,21 +8,25 @@
 use Drupal\activity_creator\Plugin\ActivityActionInterface;
 use Drupal\Core\Entity\EntityInterface;
 use Drupal\Core\Entity\EntityPublishedInterface;
+use Drupal\node\Entity\Node;
 use Drupal\node\NodeInterface;
 use Drupal\group\Entity\GroupContent;
 
 /**
  * Implements hook_entity_insert().
  */
 function activity_basics_entity_insert(EntityInterface $entity): void {
-  _activity_basics_entity_action($entity, 'create_entitiy_action');
 
   // Check if the entity created with "published" status.
   if ($entity instanceof EntityPublishedInterface) {
     if ($entity->isPublished()) {
+      _activity_basics_entity_action($entity, 'create_entitiy_action');
       _activity_basics_entity_action($entity, 'publish_entity_action');
     }
   }
+  else {
+    _activity_basics_entity_action($entity, 'create_entitiy_action');
+  }
```

Comment:
I would suggest to do early return here, so we can get rid of the else condition

```
  // Check if the entity created with "published" status.
  if ($entity instanceof EntityPublishedInterface) {
    if ($entity->isPublished()) {
      _activity_basics_entity_action($entity, 'create_entitiy_action');
      _activity_basics_entity_action($entity, 'publish_entity_action');
    }

    return;
  }

  _activity_basics_entity_action($entity, 'create_entitiy_action');
```

Reply:
tested and followed your suggestion, but can not push any more ,,,
git.exe push --progress "origin" bug_3315676_activity_for_unpublished_node:bug_3315676_activity_for_unpublished_node
remote: Permission to goalgorilla/open_social.git denied to slowflyer.
fatal: unable to access 'https://github.com/goalgorilla/open_social.git/': The requested URL returned error: 403


git did not exit cleanly (exit code 128) (4547 ms @ 05.01.2023 12:30:58)